src/main/kotlin/com/milaboratory/mixcr/cli/CommandBuildLibrary.kt

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (c) 2014-2023, MiLaboratories Inc. All Rights Reserved
+ *
+ * Before downloading or accessing the software, please read carefully the
+ * License Agreement available at:
+ * https://github.com/milaboratory/mixcr/blob/develop/LICENSE
+ *
+ * By downloading or accessing the software, you accept and agree to be bound
+ * by the terms of the License Agreement. If you do not want to agree to the terms
+ * of the Licensing Agreement, you must not download or access the software.
+ */
+package com.milaboratory.mixcr.cli
+
+import com.milaboratory.app.InputFileType
+import com.milaboratory.app.ValidationException
+import com.milaboratory.app.logger
+import io.repseq.core.Chains
+import io.repseq.core.GeneFeature
+import io.repseq.core.GeneType
+import io.repseq.core.GeneType.Constant
+import io.repseq.core.GeneType.Diversity
+import io.repseq.core.GeneType.Joining
+import io.repseq.core.GeneType.Variable
+import io.repseq.core.VDJCLibraryRegistry
+import io.repseq.dto.VDJCDataUtils.writeToFile
+import picocli.CommandLine.ArgGroup
+import picocli.CommandLine.Command
+import picocli.CommandLine.Option
+import picocli.CommandLine.Parameters
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+
+@Command(
+    description = ["Build custom reference library"]
+)
+class CommandBuildLibrary : MiXCRCommandWithOutputs() {
+
+    interface GenesFasta {
+        var pathOps: Path?
```

Comment:
may be notNull and use lateinit in overriden field?

Reply:
lateinit can't be used with custom setter, which is needed to validate fasta file type

Reply:
You can override `validate()` if you want to and move validation of extension there

Reply:
done